src/test/regress/expected/foreign_key_restriction_enforcement.out

```diff
@@ -0,0 +1,1089 @@
+	UPDATE on_update_fkey_table SET value_1 = 15;
+ROLLBACK;
+-- case 2.4: UPDATE to a reference table is followed by multiple router UPDATEs
+BEGIN;
+	UPDATE referece_table SET id = 101 WHERE id = 99;
+DEBUG:  switching to sequential query execution mode
+DETAIL:  Reference relation "referece_table" is modified, which might have cascaded to hash distributed relations via foreign keys. Any parallel modification to those hash distributed relations in the same transaction can only be executed in sequential query execution mode
+	UPDATE on_update_fkey_table SET value_1 = 101 WHERE id = 1;
+	UPDATE on_update_fkey_table SET value_1 = 101 WHERE id = 2;
+	UPDATE on_update_fkey_table SET value_1 = 101 WHERE id = 3;
+	UPDATE on_update_fkey_table SET value_1 = 101 WHERE id = 4;
+ROLLBACK;
+-- case 2.5: UPDATE to a reference table is followed by a DDL that touches fkey column
+BEGIN;
+	UPDATE referece_table SET id = 101 WHERE id = 99;
+DEBUG:  switching to sequential query execution mode
+DETAIL:  Reference relation "referece_table" is modified, which might have cascaded to hash distributed relations via foreign keys. Any parallel modification to those hash distributed relations in the same transaction can only be executed in sequential query execution mode
+	ALTER TABLE on_update_fkey_table ALTER COLUMN value_1 SET DATA TYPE bigint;
+DEBUG:  rewriting table "on_update_fkey_table"
+DEBUG:  building index "on_update_fkey_table_pkey" on table "on_update_fkey_table"
+DEBUG:  validating foreign key constraint "fkey"
+ROLLBACK;
+-- case 2.6: UPDATE to a reference table is followed by an unrelated DDL
+BEGIN;
+	UPDATE referece_table SET id = 101 WHERE id = 99;
+DEBUG:  switching to sequential query execution mode
+DETAIL:  Reference relation "referece_table" is modified, which might have cascaded to hash distributed relations via foreign keys. Any parallel modification to those hash distributed relations in the same transaction can only be executed in sequential query execution mode
+	ALTER TABLE on_update_fkey_table ADD COLUMN value_1_X INT;
+ROLLBACK;
+-- case 2.7: UPDATE to a reference table is followed by COPY
+BEGIN;
+	UPDATE referece_table SET id = 101 WHERE id = 99;
+DEBUG:  switching to sequential query execution mode
+DETAIL:  Reference relation "referece_table" is modified, which might have cascaded to hash distributed relations via foreign keys. Any parallel modification to those hash distributed relations in the same transaction can only be executed in sequential query execution mode
+	COPY on_update_fkey_table FROM STDIN WITH CSV;
+ERROR:  cannot execute parallel COPY on relation "on_update_fkey_table" since it holds conflicting locks via foreign keys on reference relation "referece_table" due to a previous "DML" access in the same transaction 
```

Comment:
I think we can still be a bit more informative and provide a hint, e.g.
```
ERROR: cannot execute COPY on relation "on_update_fkey_table, since it has a foreign key to "referece_table" and "referece_table" has been modified in this transaction
DETAIL: COPY to a distributed table uses a separate set of connections which will not be able to see the uncommitted changes to the reference table.
HINT: Perform the COPY in a separate transaction
```